### PR TITLE
fix: improve uuid mocking

### PR DESCRIPTION
**Pull request by savutsang**
DS-894

The mock now generate different uuid values in a consistent manner and they appear correctly in snapshots.
The mock is applied automatically for all tests.

### CODE REVIEW

File: packages/react/test/setup.ts [hunk 1]
```diff
@@ -17,3 +17,15 @@ jest.mock('react', () => ({
     ...jest.requireActual('react'),
     useLayoutEffect: jest.requireActual('react').useEffect,
 }));
+
+{
+    let counter = 0;
+    jest.mock('../src/utils/uuid', () => ({
+        // eslint-disable-next-line no-plusplus
```
Comment (pylafleur) [inline packages/react/test/setup.ts:24]: Oh, really?

**Review by pylafleur — APPROVED** [review]
It's nice to remove all those `undefined` from the snapshots! Good job!

File: packages/react/test/setup.ts (hunk 1 above)
Comment (savutsang) [reply packages/react/test/setup.ts:24]: Yep I agree this rule must be disabled. 
I never saw anyone making the mistake as described: https://eslint.org/docs/latest/rules/no-plusplus.

Comment (pylafleur) [reply packages/react/test/setup.ts:24]: Yeah, that rule should have been `no-space-around-plusplus` 😄

**Review by kbeaulieu — APPROVED** [review]

**Pull request merged**